include/deal.II/fe/mapping_fe_field.h

```diff
@@ -32,81 +32,94 @@ DEAL_II_NAMESPACE_OPEN
- *    MappingFE<dim,spacedim> map(eulerq, dhq, mask);
- *    map.update_euler_vector_using_triangulation(eulerq);
+ *    MappingFEField<dim,spacedim> map(eulerq, dhq, mask);
  * @endcode
-
-
  *
  * @author Luca Heltai, Marco Tezzele 2013, 2015
  */
 template <int dim, int spacedim=dim,
-          class DH=DoFHandler<dim,spacedim>,
-          class VECTOR=Vector<double> >
-class MappingFE : public Mapping<dim,spacedim>
+          class VECTOR=Vector<double>,
+          class DH=DoFHandler<dim,spacedim> >
+class MappingFEField : public Mapping<dim,spacedim>
 {
 public:
   /**
    * Constructor. The first argument is a VECTOR that specifies the
    * transformation of the domain from the reference to the current
-   * configuration. This is filled calling the method
-   * update_euler_vector_using_triangulation.
+   * configuration.
+   *
+   * In general this class decouples geometry from topology, allowing
+   * users to define geometries which are only topologically
+   * equivalent to the underlying Triangulation, but which may
+   * otherwise be arbitrary. Differently from what happens in
+   * MappingQEulerian, the FiniteElement field which is passed to the
+   * constructor is interpreted as an absolute geometrical
+   * configuration, therefore one has to make sure that the
+   * euler_vector actually represents a valid geometry (i.e., one with
+   * no inverted cells, or with no zero-volume cells).
+   *
+   * If the underlying FiniteElement is a system of FE_Q(), and
+   * euler_vector is initialized using
+   * VectorTools::get_position_vector(), then this class is in all
+   * respects identical to MappingQ().
+   *
+   * The optional ComponentMask argument can be used to specify what
+   * components of the FiniteElement to use for the geometrical
+   * transformation. If no mask is specified at construction time,
+   * then a default one is used, which makes this class works in the
+   * same way of MappingQEulerian(), i.e., the first spacedim
+   * components of the FiniteElement are assumed to represent the
+   * geometry of the problem.
+   *
+   * Notice that if a mask is specified, it has to match in size the
+   * underlying FiniteElement, and it has to have exactly spacedim
+   * non-zero elements, indicating the components (in order) of the
+   * FiniteElement which will be used for the geometry.
+   *
+   * If an incompatible mask is passed, an exception is thrown.
    */
-  MappingFE (const VECTOR  &euler_vector,
-             const DH      &euler_dof_handler,
-             const ComponentMask mask=ComponentMask());
+  MappingFEField (const VECTOR  &euler_vector,
+                  const DH      &euler_dof_handler,
+                  const ComponentMask mask=ComponentMask());
 
   /**
    * Copy constructor. Performs a deep copy, i.e. duplicates what #tensor_pols
    * points to instead of simply copying the #tensor_pols pointer as done by a
    * default copy constructor.
    */
```

Comment:
What does tensor_pols refer  here? It's not anything that's part of the public interface, I suppose?